src/KeyedDistributions.jl

```
@@ -54,6 +54,12 @@ for T in (:Distribution, :Sampleable)
                     "lengths of key vectors $key_lengths must match " *
                     "size of distribution $(_size(d))"
                 ))
+                if d isa Distribution && mean(d) isa KeyedArray && !(axiskeys(mean(d)) == keys)
```

won't we run into the same issue with `cov`? or any other function that might be return a KeyedArray?

Would it be better to `rekey` the underlying parameters and mask the exception?

---

It looks like we won't need to look at `cov` actually
```julia
julia> mvn = MvNormal(ka, sigma);

julia> mean(mvn)
1-dimensional KeyedArray(NamedDimsArray(...)) with keys:
↓   t ∈ 3-element Vector{String}
And data, 3-element Vector{Float64}:
 ("a")  0.3981265148452625
 ("b")  0.23250652731094346
 ("c")  0.09181862970601418

julia> cov(mvn)
3×3 Matrix{Float64}:
 0.0702741  0.0       0.0
 0.0        0.689994  0.0
 0.0        0.0       0.800233

# which is because of

julia> cholesky(sigma)
Cholesky{Float64, Matrix{Float64}}
U factor:
3×3 UpperTriangular{Float64, Matrix{Float64}}:
 0.265093  0.0       0.0
  ⋅        0.830659  0.0
  ⋅         ⋅        0.894558
```



and yes this 

>Would it be better to rekey the underlying parameters and just mask the exception altogether?

is a good idea

---

>Would it be better to rekey the underlying parameters and just mask the exception altogether?

I think this is actually quite hard. I think we would have to recreate the underlying distribution with the `keyless_unname` arrays. The issue is that constructors for distributions differ. We could possibly do something nasty like `keyless_unname.(fieldnames...)` and call the default constructor but it sounds like asking for trouble.

Does that leave us with throwing an exception? Or should we simply continue ostriching the issue of mismatching inner and outer keys?